index.js

```diff
@@ -873,6 +875,36 @@ class KeyringController extends EventEmitter {
       );
     }
   }
+
+  async _newKeyring(type, data) {
+    const keyringBuilder = this.getKeyringClassForType(type);
+
+    if (!keyringBuilder) {
+      return undefined;
+    }
+
+    const keyring = keyringBuilder(data);
+
+    if (keyring.init) {
+      await keyring.init();
+    }
+
+    await keyring.deserialize(data);
+
+    return keyring;
+  }
+}
+
```

Comment:
There are many instances in which we try to access the `type` from the static class. "Luckily", javascript allows a function to also have properties.

By exporting this helper method we can make sure that whenever we create a keyring builder function, the type is added as well.

Reply:
Also, since we are always calling `deserialize` after instantiating the keyring, I'm just passing the bridge to the constructor.

`deserialize` always seems return a promise, so it shouldn't be called in the constructor.

Reply:
If we're changing the "Keyring" API in a non-backwards-compatible way, I'd prefer that we start using an object for constructor options rather than parameters (i.e. the "options bag" pattern, passing in `{ bridge }` rather than `bridge`). This pattern is easier to read, and makes it easy to add additional options without breaking changes.

Reply:
Fair enough. I've made the change to the options bag pattern directly in `eth-trezor-keyring` PR. There's no longer any reference here as the new simple builder factory does not concern itself with bridges anymore.